target/linux/realtek/dts/rtl8380_netgear_gigabit.dtsi

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "rtl838x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "realtek,rtl838x-soc";
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		mode {
+			label = "reset";
+			gpios = <&gpio0 24 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gpio0 {
+	indirect-access-bus-id = <0>;
+};
+
+&spi0 {
+	status = "okay";
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```

Comment:
Can this go faster?

Reply:
Not sure; I have seen multiple flash chip vendors for these platforms which probably support different frequencies.  I'll check the data sheet for the one in my switch.

Reply:
From the datasheet: "SPI clock frequencies of W25Q256JV of up to 133MHz are supported".  I did see the response from @svanheule about how this property isn't currently used.  I'll update the speed for now and can remove it if requested.

Reply:
Based on @svanheule 's response, I'd be fine with keeping this at 10 MHz ...

Reply:
Honestly I see no reason to revert to 10MHz.  If the node truly is unused then its value is of no consequence.  Both chips known to be in use on this platform -- Macronix MX25L25635F and Winbond W25Q256JV -- support clock frequencies up to 133MHz.  If nothing else than this serves as documentation for future code archeologists.

Reply:
Using untested `spi-max-frequency` is not a good practice, as shown in 9de6bc872af5 and ebc090e420d1.
The spec number is for ideal case; the max frequency on the specific board might be much lower.

Reply:
@mans0n thank you for the review.  Upon closer inspection of the datasheet it turns out that the max frequency should probably be limited to 50MHz, like the other platforms in the referenced commit.  The read data instruction is the limiting factor.

Reply:
I've looked at the binding docs for "jedec,spi-nor", and it turns out spi-max-frequency is (always) required for this node, even if the parent SPI controller does not honour it. 50MHz will _probably_ be fine, although we'll only find out with testing.